tools/tasks/seed/check.tools.ts

```diff
@@ -0,0 +1,30 @@
+import * as util from 'gulp-util';
+import * as runSequence from 'run-sequence';
+
+import { join } from 'path';
+
+let fs = require('fs');
+
+
+// if gulpfile.ts has been compiled then we need to rebuild the toolchain
+
+export = (done: any) => {
+
+  let checkFile = join(process.cwd(), 'tools', 'config.js');
+
+  // need to require the build.toolchain task as it won't be able to run after we run clear.files
+  let buildTools = require('./build.tools');
+
+  try {
+    fs.accessSync(checkFile, fs.F_OK);
+    util.log('Gulpfile has previously been compiled, rebuilding toolchain');
+    runSequence('clean.tools');
```

Comment:
I'm not sure I understand what issue you hit here. I'm concerned of potential race condition where you remove files which are built by the current invocation of `build.tools` (in this case this is not going to happen because we have a call `del.sync` but it still seems a bit scary because it's easy to mess this up, it has some indirect coupling).